syntax = "proto2";
package protos;

import "easy_rec/python/protos/dnn.proto";
import "easy_rec/python/protos/simi.proto";


message RecallTower {
  required DNN dnn = 1;
};


message MultiTowerRecall {
    required RecallTower user_tower = 1;
    required RecallTower item_tower = 2;
    required float l2_regularization = 3 [default = 1e-4];
    required DNN final_dnn = 4;
    required bool ignore_in_batch_neg_sam = 10 [default = false];
}
